bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java

```diff
@@ -121,25 +135,56 @@ public BitGoClientImpl(String token, String baseUrl, boolean unsafe) {
         }
         final Map<String, Object> data = new HashMap<>();
         data.put("recipients", addr);
-        if (message != null) {
-            data.put("message", message);
-        }
-        if (sequenceId != null) {
-            data.put("sequenceId", sequenceId);
-        }
-        if (fee != null) {
-            data.put("fee", fee.longValue());
-        }
-        if (feeTxConfirmTarget != null) {
-            data.put("feeTxConfirmTarget", feeTxConfirmTarget);
-        }
-        if (minConfirms > 0) {
-            data.put("minConfirms", minConfirms);
+        data.put("walletPassphrase", walletPass);
+        data.put("sequenceId", sequenceId);
+
+        // Check for optional parameters or default them to null
+        if (optionalParameters != null) {
+            Object message = optionalParameters.containsKey("message") ? optionalParameters.get("message") : null;
+            Object fee = optionalParameters.containsKey("fee") ? optionalParameters.get("fee") : null;
+            Object feeTxConfirmTarget = optionalParameters.containsKey("feeTxConfirmTarget") ?
+                    optionalParameters.get("feeTxConfirmTarget") : null;
+            Object minConfirms = optionalParameters.containsKey("minConfirms") ? optionalParameters.get("minConfirms") : null;
+            Object enforceMinConfirmsForChange = optionalParameters.containsKey("enforceMinConfirmsForChange") ?
+                    optionalParameters.get("enforceMinConfirmsForChange") : null;
```

Comment:
We also need to support optionalParameters not in their documentation.

Can we add all the optionalParameters to the `data` hashmap?

Reply:
Ok, sure, just question: Add the whole optionalParameters map after validating the ones we know instead of adding them one by one as I did? Or add the ones known like this and add the map in case there are remaining unknown parameters to us?

Reply:
> Does TestClientV2 compile? Looks like a compilation error here:
> 
> https://github.com/bitsoex/bitso-bitgo-java-api/blob/bd5181d41b4b94dd97f9ffba81bbb551aaefd7cb/bitgo-java-api/src/test/java/com/bitso/bitgo/util/TestClientV2.java#L116

Really? Crap, I just did a build and everything was alright. Which error did you get?

Reply:
> Ok, sure, just question: Add the whole optionalParameters map after validating the ones we know instead of adding them one by one as I did? Or add the ones known like this and add the map in case there are remaining unknown parameters to us?

These two options sound the same to me?  If we validate the optional parameters that we know, and add any unknown ones, then i think we're good.

Reply:
> > Does TestClientV2 compile? Looks like a compilation error here:
> > https://github.com/bitsoex/bitso-bitgo-java-api/blob/bd5181d41b4b94dd97f9ffba81bbb551aaefd7cb/bitgo-java-api/src/test/java/com/bitso/bitgo/util/TestClientV2.java#L116
> 
> Really? Crap, I just did a build and everything was alright. Which error did you get?

Note it's the test, not the src code, that isn't compiling for me.  That line I referenced fails b/c it cannot `Cannot resolve method 'isEmpty' on 'Optional'`

Reply:
Ahh nevermind, that error was my fault - my project was on jdk8.  When I move to jdk11 it's ok

Reply:
> > Ok, sure, just question: Add the whole optionalParameters map after validating the ones we know instead of adding them one by one as I did? Or add the ones known like this and add the map in case there are remaining unknown parameters to us?
> 
> These two options sound the same to me? If we validate the optional parameters that we know, and add any unknown ones, then i think we're good.

haha no, my bad, my question was wrong, I meant if validate and then send to Bitgo a HashMap either with all the optionals or just with the unknowns, but as I check this would kill bitgo as it won't know what to do with a map of parameters. So I'll leave the validation but also loop through all the keys and add them to the data. And I guess it's on the user if a bad parameter is sent.